EasyPost.Tests/UserTest.cs

```diff
@@ -1,102 +1,124 @@
 ﻿using System.Collections.Generic;
 using System.Threading.Tasks;
-using Microsoft.VisualStudio.TestTools.UnitTesting;
+using EasyPost.Models.API;
+using EasyPost.Utilities.Annotations;
+using Xunit;
 
 namespace EasyPost.Tests
 {
-    [TestClass]
-    public class UserTest
+    public class UserTest : UnitTest
     {
-        private static string _userId = null;
-
-        private TestUtils.VCR _vcr;
-
-        [TestInitialize]
-        public void Initialize()
-        {
-            _vcr = new TestUtils.VCR("user", TestUtils.ApiKey.Production);
-        }
-
-        [TestCleanup]
-        public async Task Cleanup()
-        {
-            if (_userId != null)
+        public UserTest() : base("user", TestUtils.ApiKey.Production) =>
+            CleanupFunction = async id =>
             {
                 try
                 {
-                    User retrievedUser = await User.Retrieve(_userId);
+                    User retrievedUser = await Client.User.Retrieve(id);
                     await retrievedUser.Delete();
-                    _userId = null;
+                    return true;
                 }
                 catch
                 {
-                    // in case we try to delete something that's already been deleted
+                    // trying to delete something that doesn't exist, pass
+                    return false;
                 }
-            }
-        }
+            };
 
-        private static async Task<User> RetrieveMe()
+        #region CRUD Operations
+
+        [Fact]
+        [CrudOperations.Create]
+        public async Task TestCreate()
         {
-            return await User.RetrieveMe();
+            UseVCR("create");
+
+            User user = await CreateUser();
+
+            Assert.IsType<User>(user);
+            Assert.StartsWith("user_", user.id);
+            Assert.Equal("Test User", user.name);
         }
 
-        private static async Task<User> CreateUser()
+        [Fact]
+        [CrudOperations.Create]
+        public async Task TestUpdateBrand()
         {
-            User user = await User.Create(new Dictionary<string, object>
+            UseVCR("update_brand");
+
+            User user = await CreateUser();
+
+            string color = "#123456";
+            Brand brand = await user.UpdateBrand(new Dictionary<string, object>
             {
                 {
-                    "name", "Test User"
+                    "color", color
                 }
             });
-            _userId = user.id; // trigger deletion after test
-            return user;
+
+            Assert.IsType<Brand>(brand);
+            Assert.StartsWith("brd_", brand.id);
+            Assert.Equal(color, brand.color);
         }
 
-        // This endpoint returns the child user keys in plain text, do not run this test.
-        [TestMethod]
-        public async Task TestCreate()
+        [Fact]
+        [CrudOperations.Read]
+        public async Task TestAllApiKeys()
```

Comment:
Issue: Please move this test into its own test file since it's on an ApiKey object technically.

Reply:
Resolved.